File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int sum=0;
        sort(people.begin(),people.end());
        int s=0;
        int c=0;
        int e=people.size()-1;
        while(s<=e){
            if(people[s]+people[e]<=limit){
                c++;
                s++;
                e--;
            }else if(people[s]+people[e]>limit){
                c++;
                e--;
            }
        }  
        return c; 
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int n = people.length, ans = 0, i= 0, sum=0, boats=0,j=n-1;
        while(i<=j){
            if(people[i]+people[j]<=limit){
                boats++;
                i++;
                j--;
            }else if(people[i]+people[j]>limit){
                boats++;
                j--;
            }
        }
        return boats;
    }
}
// 1 2 2 3  3